### PR TITLE
Refactor Accordion components

### LINKED ISSUE #5247: New Accordion subcomponents
Instead of using the Collapse component for the accordion structure, we need to introduce new Accordion specific components

- AccordionItem
- AccordionHeader
- AccordionBody

### PR DESCRIPTION
**Pull request by stsrki**
Closes #5247

Introduced new Accordion subcomponets

- `AccordionItem`, and
- `AccordionHeader`, and
- `AccordionBody`

Reason is that using the Collapse was leaking too much of logic between accordion and collapse, which are not that much directly connected. This PR should solve this because we now have dedicated component just for accordion. In the future we should strip collapse from accordion completely #5248.

### CODE REVIEW

File: Source/Blazorise/Components/Accordion/AccordionToggle.razor.cs [hunk 1]
```diff
@@ -41,6 +43,11 @@ protected async Task ClickHandler( MouseEventArgs eventArgs )
             await ParentCollapse.Toggle();
         }
 
+        if ( ParentAccordionItem is not null )
```
Comment (David-Moreira) [inline Source/Blazorise/Components/Accordion/AccordionToggle.razor.cs:46]: If might be a little more correct to do an else if. Although it's one of those conditions where you already expect only one of them is to be executed.

File: Source/Blazorise.FluentUI2/Components/AccordionToggle.razor [hunk 2]
```diff
@@ -6,5 +6,5 @@
     @ChildContent
 </button>
 @code {
-    string AriaExpandedString => CollapseVisible ? "true" : "false";
+    string AriaExpandedString => AccordionItemVisible ? "true" : "false";
```
Comment (David-Moreira) [inline Source/Blazorise.FluentUI2/Components/AccordionToggle.razor:9]: Do we not need to check for Collapse?

File: Source/Blazorise/Components/Accordion/AccordionToggle.razor.cs [hunk 3]
```diff
@@ -76,6 +88,24 @@ public bool CollapseVisible
         }
     }
 
+    /// <summary>
+    /// Gets or sets the content visibility.
+    /// </summary>
+    [CascadingParameter( Name = "AccordionItemVisible" )]
+    public bool AccordionItemVisible
```
Comment (David-Moreira) [inline Source/Blazorise/Components/Accordion/AccordionToggle.razor.cs:95]: Didn't we use a state mechanism for this things before?

File: Source/Blazorise/Components/Accordion/Accordion.razor.cs [hunk 4]
```diff
@@ -56,6 +58,35 @@ internal bool IsLastInAccordion( Collapse collapse )
...
+    {
+        if ( accordionItems is not null && accordionItems.IndexOf( accordionItem ) == 0 )
+            return true;
+
+        return false;
+    }
+
+    internal bool IsLastInAccordion( AccordionItem accordionItem )
+    {
+        if ( accordionItems is not null && accordionItems.IndexOf( accordionItem ) == accordionItems.Count - 1 )
```
Comment (David-Moreira) [inline Source/Blazorise/Components/Accordion/Accordion.razor.cs:84]: I doubt it, but could this have a dumb bug like:
accordionItems being empty, 
which means  
**accordionItems.IndexOf( accordionItem )** => **-1**
**accordionItems.Count - 1** would also lead to **-1**

File: Source/Blazorise.Material/MaterialClassProvider.cs [hunk 5]
```diff
@@ -37,5 +37,9 @@ public class MaterialClassProvider : Bootstrap.BootstrapClassProvider
 
     public override string BarItemHasDropdown( BarMode mode ) => mode == Blazorise.BarMode.Horizontal ? "dropdown" : null;
 
+    public override string AccordionToggle() => "btn btn-link btn-block text-left";
+
+    public override string AccordionToggleCollapsed( bool collapsed ) => collapsed ? null : "collapsed";
```
Comment (David-Moreira) [inline Source/Blazorise.Material/MaterialClassProvider.cs:42]: is this right? If collapsed null otherwise "collapsed"?

**Review by David-Moreira — APPROVED** [review]
LGTM, more coherent api.

File: Source/Blazorise/Components/Accordion/Accordion.razor.cs (hunk 4 above)
Comment (stsrki) [reply Source/Blazorise/Components/Accordion/Accordion.razor.cs:84]: If the accordionItems are empty, then it is already invalid. I would leave it like this. I don't want to introduce another bug by accident. If anyone complains in the future, we will change it.

File: Source/Blazorise.Material/MaterialClassProvider.cs (hunk 5 above)
Comment (stsrki) [reply Source/Blazorise.Material/MaterialClassProvider.cs:42]: It is. We send it a `Visible`  parameter, and in here we name it `collapsed`. It's inverted so there's a confusion.

File: Source/Blazorise/Components/Accordion/AccordionToggle.razor.cs (hunk 3 above)
Comment (stsrki) [reply Source/Blazorise/Components/Accordion/AccordionToggle.razor.cs:94]: For Accordion no. We keep it simple until we don't :)

**Pull request merged**